includes/class-wc-payments-upe-checkout.php

```diff
@@ -183,12 +183,26 @@ public function get_enabled_payment_method_config() {
 					]
 				),
 				'icon'                 => $payment_method->get_icon(),
+				'showSaveOption'       => $this->should_upe_payment_method_show_save_option( $payment_method_id ),
 			];
 		}
 
 		return $settings;
 	}
 
+	/**
+	 * Checks if the save option for a payment method should be displayed or not.
+	 *
+	 * @param string $payment_method_id Payment method ID, e.g. sepa_debit or bancontact.
+	 * @return bool - True if the payment method is SEPA and the saved cards feature is enabled for the gateway and there is no subscription item in the cart, false otherwise.
+	 */
+	private function should_upe_payment_method_show_save_option( $payment_method_id ) {
+		if ( Payment_Method::SEPA === $payment_method_id ) {
+			return $this->gateway->is_saved_cards_enabled() && ! $this->gateway->is_subscription_item_in_cart();
+		}
+		return false;
+	}
```

Comment:
```suggestion
	 * @return bool - True if the payment method is reusable and the saved cards feature is enabled for the gateway and there is no subscription item in the cart, false otherwise.
	 */
	private function should_upe_payment_method_show_save_option( $payment_method_id ) {
		if ( $this->gateway->wc_payments_get_payment_method_by_id( $payment_method_id )->is_reusable() ) {
			return $this->gateway->is_saved_cards_enabled() && ! $this->gateway->is_subscription_item_in_cart();
		}
		return false;
	}
```

As I mentioned above, the `is_reusable` flag inside each payment method determines whether a payment method type can be tokenized. Right now, we only support saving SEPA payment methods, but in the future we'll add more. My understanding is that Stripe supports tokenizing several of these payment method types already. If we decide to move the logic to the back-end here, it would be good to future proof the function rather that specifying SEPA. There are probably several other areas of WCPay that would benefit from the same idea.

Reply:
Thank you for explaining the "business" background of the `is_reusable` flag - I didn't know it reflects the subset of payment methods, which can be saved. I checked how the method works and confirmed that it's the equivalent of the SEPA subset.

This can (even should) replace `Payment_Method::SEPA === $payment_method_id` to avoid any changes in the future when the set of tokenized methods is updated. This is a super important note from you, and thanks for it! Fixed. 

> 💡 instead of `$this->gateway->wc_payments_get_payment_method_by_id( $payment_method_id )->is_reusable()` I'll provide `$payment_method` as it's already available.

Reply:
>This can (even should) replace Payment_Method::SEPA === $payment_method_id to avoid any changes in the future when the set of tokenized methods is updated. This is a super important note from you, and thanks for it! Fixed.

Perfect!  Thank you 🙌. 

>I'll provide $payment_method as it's already available.

Ah, nice catch! Looks great.